### DISCUSSION TITLE
New GraalVM version scheme and Mandrel sticking to the old one

### DISCUSSION THREAD

**Discussion by zakkak** [graalvm/mandrel#520] [General]
Starting with the latest GraalVM release, GraalVM adopted a new version scheme which essentially means that GraalVM inherits the version of the JDK it's based on, e.g. _GraalVM for JDK 20 Community 20.0.1_. Mandrel, however, still follows the YY.X version possibly making it hard for users to map upstream GraalVM versions to the corresponding Mandrel version.

To fix (or mitigate) this, Quarkus builder images are tagged both with the new and the old version scheme:

![image](https://github.com/graalvm/mandrel/assets/1435395/7b114220-815c-4cf3-a72e-7bf3141204af)

However, since we don't do the same with our GitHub releases, other tools (and users) might find it hard to do the mapping. 

For instance, with the new release of GraalVM and the new versioning scheme SDKMAN introduced two new `java` candidates, `graalce` for _GraalVM CE_ and `graal` for _Oracle GraalVM_. In the past there was only the `grl` candidate for GraalVM CE releases that could easily be mapped to the corresponding Mandrel version, but appears to no longer be used for new releases.

FWIW, the GraalVM versions on SDKMAN look like this:

```
 GraalVM       |     | 22.3.r17     | grl     |            | 22.3.r17-grl        
               |     | 22.3.r11     | grl     |            | 22.3.r11-grl        
               |     | 22.3.2.r17   | grl     |            | 22.3.2.r17-grl      
               |     | 22.3.2.r11   | grl     |            | 22.3.2.r11-grl      
               |     | 22.3.1.r19   | grl     |            | 22.3.1.r19-grl      
               |     | 22.3.1.r17   | grl     |            | 22.3.1.r17-grl      
...
 GraalVM CE    |     | 20.0.1       | graalce |            | 20.0.1-graalce      
               |     | 17.0.7       | graalce |            | 17.0.7-graalce      
 GraalVM Oracle|     | 20.0.1       | graal   |            | 20.0.1-graal        
               |     | 17.0.7       | graal   |            | 17.0.7-graal       
```

So I wonder if it would make sense for the Mandrel repository to introduce jdk-XX tags along with the YY.X tags.

cc @jerboaa @Karm @galderz

**Comment by galderz** [comment]
> So I wonder if it would make sense for the Mandrel repository to introduce jdk-XX tags along with the YY.X tags.

Hmmmm, seems to make sense, but not sure if there's any downsides to this. I guess we'd just have multiple tags for the same thing. I don't see anything majorly wrong with it at a first glance. The only issue/problem I see is documenting the mapping, but we need that anyway because we can limit ourself to the GraalVM and/or Oracle versioning scheme.

**Reply by jerboaa** [reply]
Agreed. I'm fine with additional tags (they'd probably be `<jdk-version>-<something>` so as to have equivalence of what we had with `22.3.0.0`, `22.3.0.1` etc.). FWIW, for a GraalVM distro which keeps the OpenJDK version unmodified most of the publicly visible version info is useless to relate to anything graal codebase related, so I'm not a fan of the new version scheme (as you can probably read between the lines).
